executor/simple.go

```diff
@@ -1059,21 +1062,86 @@ func (e *SimpleExec) executeSetPwd(s *ast.SetPwdStmt) error {
 	return err
 }
 
-func (e *SimpleExec) executeKillStmt(s *ast.KillStmt) error {
-	conf := config.GetGlobalConfig()
-	if s.TiDBExtension || conf.CompatibleKillQuery {
-		sm := e.ctx.GetSessionManager()
-		if sm == nil {
-			return nil
+func (e *SimpleExec) executeKillStmt(ctx context.Context, s *ast.KillStmt) error {
+	sm := e.ctx.GetSessionManager()
+	if sm == nil {
+		return nil
+	}
+
+	if sm.ServerID() == 0 {
+		// `sm.ServerID() == 0` indicates serverID allocation by PD would be not working.
+		// Degrade to behavior as early versions.
+		conf := config.GetGlobalConfig()
+		if s.TiDBExtension || conf.CompatibleKillQuery {
+			sm.Kill(s.ConnectionID, s.Query)
+		} else {
+			err := errors.New("Invalid operation. Please use 'KILL TIDB [CONNECTION | QUERY] connectionID' instead")
+			e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
 		}
-		sm.Kill(s.ConnectionID, s.Query)
-	} else {
-		err := errors.New("Invalid operation. Please use 'KILL TIDB [CONNECTION | QUERY] connectionID' instead")
+		return nil
+	}
+
+	connID, isTruncated := util.ParseGlobalConnID(s.ConnectionID)
+	if isTruncated {
+		err := errors.New("Invalid operation. ConnectionID is truncated to 32 bits")
 		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+		return nil
+	}
+
+	if connID.ServerID != sm.ServerID() {
+		if err := killRemoteConn(ctx, e.ctx, &connID, s.Query); err != nil {
+			err1 := errors.New("Kill remote connection failed: " + err.Error())
+			e.ctx.GetSessionVars().StmtCtx.AppendWarning(err1)
+		}
+	} else {
+		sm.Kill(s.ConnectionID, s.Query)
```

Comment:
There is a log for remote kill, maybe we should also add a log for local kill for consistency consideration

Reply:
We already have a log in underlying [here](https://github.com/pingcap/tidb/blob/2a8b968453520e4fcf9d6ff46c9f23b4ad23feee/server/server.go#L509). It may be redundant.